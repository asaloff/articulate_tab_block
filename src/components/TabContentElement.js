import React, { Component } from 'react';
import PropTypes from 'prop-types';

class TabContentElement extends Component {
  constructor() {
    super();
    this.state = { visible: false };

    this.makeVisible = this.makeVisible.bind(this);
  }

  componentDidMount() {
    this.makeVisible();
  }

  componentWillReceiveProps() {
    this.setState({ visible: false });
  }

  componentDidUpdate() {
    if (!this.state.visible) {
      this.makeVisible();
    }
  }

  makeVisible() {
    setTimeout(() => {
      this.setState({ visible: true });
    }, 0);
  }

  render() {
    const { content } = this.props;
    let children;
    const props = {
      className: `TabContentElement ${ this.state.visible && 'visible' }`
    };

    if (content.tag === 'img') {
      children = null;
      props.src = content.content;
    } else {
      children = content.content;
    }

    return React.createElement(content.tag, props, children);
  }
}

TabContentElement.propTypes = {
  content: PropTypes.object.isRequired
};

export default TabContentElement;
