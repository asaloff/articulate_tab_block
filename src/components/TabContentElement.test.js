import React from 'react';
import { shallow } from 'enzyme';
import TabContentElement from './TabContentElement';
import tabData from '../tabs';

describe('TabContentElement', () => {
  // console.log(tabData.tabs[0])
  let props = { content: tabData.tabs[0].content[0] };
  let tabContentElement = shallow(<TabContentElement {...props} />);

  it('renders properly', () => expect(tabContentElement).toMatchSnapshot());

  it('displays the correct paragraph text', () => {
    const expectedContent = tabData.tabs[0].content[0].content;
    expect(tabContentElement.find('p').text()).toEqual(expectedContent);
  });

  it('contains an img tag with the correct href', () => {
    let props = { content: tabData.tabs[0].content[1] };
    let tabContentElement = shallow(<TabContentElement {...props} />);

    expect(tabContentElement
      .find(`img[src="${tabData.tabs[0].content[1].content}"]`)
      .exists()
    ).toBe(true);
  });
});
