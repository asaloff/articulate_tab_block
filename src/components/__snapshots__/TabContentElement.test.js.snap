// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContentElement renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabContentElement
    content={
        Object {
            "content": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
            "tag": "p",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
      "className": "TabContentElement false",
    },
    "ref": null,
    "rendered": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
        "className": "TabContentElement false",
      },
      "ref": null,
      "rendered": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
